drracket/drracket/private/language.rkt

```diff
@@ -932,35 +915,32 @@
        (string-constant cancel)))
     
     (define (browse-callback)
-      (let ([ftf (send filename-text-field get-value)])
-        (let-values ([(base name _)
-                      (if (path-string? ftf)
-                          (split-path ftf)
-                          (values (current-directory) "" #f))])
-          (let* ([mzscheme? (currently-mzscheme-binary?)]
-                 [mode (current-mode)]
-                 [filename 
-                  (put-executable/defaults
-                   dlg
-                   base
-                   name
-                   mode
-                   (not mzscheme?)
-                   (case mode
-                     [(launcher)
-                      (if mzscheme?
-                          (string-constant save-a-mzscheme-launcher)
-                          (string-constant save-a-mred-launcher))]
-                     [(stand-alone)
-                      (if mzscheme?
-                          (string-constant save-a-mzscheme-stand-alone-executable)
-                          (string-constant save-a-mred-stand-alone-executable))]
-                     [(distribution)
-                      (if mzscheme?
-                          (string-constant save-a-mzscheme-distribution)
-                          (string-constant save-a-mred-distribution))]))])
-            (when filename
-              (send filename-text-field set-value (path->string filename)))))))
+      (define ftf (send filename-text-field get-value))
+      (let-values ([(base name _)
```

Comment:
Why didn't this let turn into a define?

Reply:
It does seem to make sense to do them all at once when there are a bunch of them together. At least, that's what I usually do when I'm doing this kind of thing by hand; I'll do all of them if I do any of them in a particular block.

Reply:
Filed jackfirth/resyntax#284.